plugins/BEdita/Core/src/Model/Table/FoldersTable.php

```diff
@@ -150,4 +152,27 @@ public function afterSave(Event $event, EntityInterface $entity)
         $node->parent_id = $entity->parent_id;
         $trees->saveOrFail($node);
     }
+
+    /**
+     * Finder for root folders.
+     *
+     * @param Query $query Query object instance.
+     * @return \Cake\ORM\Query
+     */
+    protected function findRoots(Query $query)
+    {
+        $subquery = TableRegistry::get('Trees')->find('list')->where(['parent_id IS NULL']);
+
+        $query->join([
+            'table' => 'trees',
+            'alias' => 'Trees',
+            'type' => 'INNER',
+            'conditions' => 'Trees.object_id = ' . $this->aliasField('id'),
+        ])
+        ->where(function (QueryExpression $exp) use ($subquery) {
```

Comment:
I think it can be simplified without using subquery as

```php
$query->join([
            'table' => 'trees',
            'alias' => 'Trees',
            'type' => 'INNER',
            'conditions' => 'Trees.object_id = ' . $this->aliasField('id'),
        ])
        ->where(function (QueryExpression $exp) {
            return $exp->isNull('Trees.parent_id');
        });
```